### PR TITLE
Sass 1247

### PR DESCRIPTION
**Pull request by BenCosfordHMRC**
Added when did you leave page
https://jira.tools.tax.service.gov.uk/browse/SASS-1247

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of master?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/controllers/employment/EmployerLeaveDateControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,927 @@
...
+            inputFieldValueCheck(nowDatePlusOne.getYear.toString, Selectors.yearSelector)
+            buttonCheck(expectedButtonText, continueButtonSelector)
+            formPostLinkCheck(continueLink, continueButtonFormSelector)
+            welshToggleCheck(user.isWelsh)
+
+            errorSummaryCheck(user.specificExpectedResults.get.futureDateError, Selectors.daySelector)
+            errorAboveElementCheck(user.specificExpectedResults.get.futureDateError, Some("amount"))
+          }
+          "the date is before the start date" which {
+            val nowDatePlusOne = LocalDate.now().plusDays(1)
```
Comment (ina-tu) [inline it/controllers/employment/EmployerLeaveDateControllerISpec.scala:885]: nowDatePlusOne isn't used - can be removed

File: it/controllers/employment/EmployerLeaveDateControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,927 @@
...
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.http.HeaderNames
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import play.api.libs.ws.WSResponse
+import utils.{EmploymentDatabaseHelper, IntegrationTest, ViewHelpers}
+
+import java.time.LocalDate
+
+//scalastyle:off
```
Comment (ina-tu) [inline it/controllers/employment/EmployerLeaveDateControllerISpec.scala:31]: is scalastyle:off still needed?

File: it/controllers/employment/EmployerLeaveDateControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,927 @@
...
+  }
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, SpecificExpectedResults]] = {
+    Seq(UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+      UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+      UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+      UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY)))
+  }
+
+  object CyaModel {
```
Comment (ina-tu) [inline it/controllers/employment/EmployerLeaveDateControllerISpec.scala:191]: CyaModel object isn't used - can be removed

**Review by ina-tu — CHANGES_REQUESTED** [review]

File: app/models/mongo/EmploymentCYAModel.scala [hunk 4]
```diff
@@ -34,11 +34,20 @@ case class EmploymentDetails(employerName: String,
                              currentDataIsHmrcHeld: Boolean) {
 
   val isFinished: Boolean = {
+
+    val cessationSectionFinished = {
+      cessationDateQuestion match {
+        case Some(true) => true
+        case Some(false) => cessationDate.isDefined
+        case None => false
+      }
```
[the remark this answers is not in the document]
Comment (BenCosfordHMRC) [reply app/models/mongo/EmploymentCYAModel.scala:43]: removed

File: it/controllers/employment/EmployerLeaveDateControllerISpec.scala (hunk 1 above)
Comment (BenCosfordHMRC) [reply it/controllers/employment/EmployerLeaveDateControllerISpec.scala]: removed

File: it/controllers/employment/EmployerLeaveDateControllerISpec.scala (hunk 2 above)
Comment (BenCosfordHMRC) [reply it/controllers/employment/EmployerLeaveDateControllerISpec.scala:31]: kept it to stop the highlighting, scalastyle being turned off is much more of concern in the actual code imo

File: it/controllers/employment/EmployerLeaveDateControllerISpec.scala (hunk 3 above)
Comment (BenCosfordHMRC) [reply it/controllers/employment/EmployerLeaveDateControllerISpec.scala]: removed

**Review by ina-tu — APPROVED** [review]

**Pull request merged**